src/main/java/net/maritimeconnectivity/serviceregistry/controllers/secom/v2/SecomV2SearchServiceController.java

```diff
@@ -0,0 +1,323 @@
+import net.maritimeconnectivity.serviceregistry.models.domain.Instance;
+import net.maritimeconnectivity.serviceregistry.models.domain.enums.BooleanOperator;
+import net.maritimeconnectivity.serviceregistry.models.dto.mcp.McpCertificateDto;
+import net.maritimeconnectivity.serviceregistry.models.dto.mcp.McpEntityBase;
+import net.maritimeconnectivity.serviceregistry.models.dto.mcp.McpServiceDto;
+import net.maritimeconnectivity.serviceregistry.models.dto.secom.v2.SearchObjectResultWithCert;
+import net.maritimeconnectivity.serviceregistry.services.InstanceService;
+import net.maritimeconnectivity.serviceregistry.utils.GeometryJSONConverter;
+import net.maritimeconnectivity.serviceregistry.utils.WKTUtil;
+import org.apache.logging.log4j.util.Strings;
+import org.assertj.core.util.Arrays;
+import org.grad.secomv2.core.exceptions.SecomValidationException;
+import org.grad.secomv2.core.interfaces.SearchServiceServiceInterface;
+import org.grad.secomv2.core.models.ResponseSearchObject;
+import org.grad.secomv2.core.models.SearchFilterObject;
+import org.grad.secomv2.core.models.SearchObjectResult;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.io.ParseException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.validation.annotation.Validated;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static java.util.function.Predicate.not;
+
+/**
+ * The SECOM Discovery Service Controller.
+ *
+ * @author Nikolaos Vastardis (email: Nikolaos.Vastardis@gla-rad.org)
+ */
+@Component
+@Path("/")
+@Validated
+@Slf4j
+public class SecomV2SearchServiceController implements SearchServiceServiceInterface {
+
+    /**
+     * The Object Mapper.
+     */
+    @Autowired
+    ObjectMapper objectMapper;
+
+    /**
+     * The Instance Service.
+     */
+    @Autowired
+    InstanceService instanceService;
+
+    @Autowired(required = false)
+    MirClient mirClient;
+
+    /**
+     * Object Mapper from Domain to DTO.
+     */
+    @Autowired
+    DomainDtoMapper<Instance, SearchObjectResult> searchObjectResultMapper;
+
+    /**
+     * POST /v2/searchService : The purpose of this interface is to search for
+     * service instances to consume.
+     *
+     * @param searchFilterObject The search filter object
+     * @return the result list of the search
+     */
+    @Tag(name = "SECOM")
+    @Transactional
+    @Path(SEARCH_SERVICE_INTERFACE_PATH)
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public ResponseSearchObject searchService(@Valid SearchFilterObject searchFilterObject)  {
```

Comment:
Would it be possible to also have a GET version of this, as that is required by the specification?